modules/dists/BlockDist.chpl

```diff
@@ -952,6 +967,54 @@ proc BlockDom.dsiBuildArray(type eltType, param initElts:bool) {
   return arr;
 }
 
+proc BlockDom.doiTryBuildArray(type eltType, param initElts=true) throws {
+  const dom = this;
+  const creationLocale = here.id;
+  const dummyLBD = new unmanaged LocBlockDom(rank, idxType, strides);
+  const dummyLBA = new unmanaged LocBlockArr(eltType, rank, idxType,
+                                             strides, dummyLBD, false);
+  var locArrTemp: [dom.dist.targetLocDom]
+        unmanaged LocBlockArr(eltType, rank, idxType, strides) = dummyLBA;
+  var myLocArrTemp: unmanaged LocBlockArr(eltType, rank, idxType, strides)?;
+
+  // formerly in BlockArr.setup()
+  coforall (loc, locDomsElt, locArrTempElt)
+    in zip(dom.dist.targetLocales, dom.locDoms, locArrTemp)
+           with (ref myLocArrTemp) {
+    on loc {
+      const locSize = locDomsElt.myBlock.size;
+      var callPostAlloc = false;
+      var data = _ddata_allocate_noinit_nocheck(eltType, locSize, callPostAlloc);
+      if data == nil then
+        throw new Error("Could not allocate memory");
```

Comment:
Might be worth adding a TODO somewhere that this ``Error`` should eventually be a different type that users could more easily distinguish.